### PR TITLE
Update CoreNEURON build and testing documentation

### PR DESCRIPTION
**Pull request by iomaganaris**
**Description**

- [x] Small improvements in build documentation
- [x] Added testing documentation

**How to test this?**

Check `README.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -234,6 +234,34 @@ For other errors, please [open an issue](https://github.com/BlueBrain/CoreNeuron
 
 ## Developer Build
 
+#### Running Unit and Integration Tests
+
+As **CoreNEURON** is mostly used as a compute library of **NEURON** it needs to be incorporated with **NEURON** to test most of its functionality. Consequently its tests are included in the NEURON repository. To enable and run all the tests of **CoreNEURON** you need to add the `-DNRN_ENABLE_TESTS=ON` CMake flag in **NEURON**.
+Those tests include:
+* Unit tests of NEURON
+* Unit tests of CoreNEURON
+* Integration tests of NEURON
```
Comment (olupton) [inline README.md:243]:
```suggestion
* Integration tests comparing NEURON, CoreNEURON and reference files
```

File: README.md [hunk 2]
```diff
@@ -234,6 +234,34 @@ For other errors, please [open an issue](https://github.com/BlueBrain/CoreNeuron
...
+* Integration tests of CoreNEURON
```
Comment (olupton) [inline README.md:244]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -234,6 +234,34 @@ For other errors, please [open an issue](https://github.com/BlueBrain/CoreNeuron
...
+* Integration tests between NEURON and CoreNEURON
```
Comment (olupton) [inline README.md:245]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -234,6 +234,34 @@ For other errors, please [open an issue](https://github.com/BlueBrain/CoreNeuron
...
+cd nrn/build
+cmake .. \
+  -DNRN_ENABLE_CORENEURON=ON \
+  -DNRN_ENABLE_INTERVIEWS=OFF \
+  -DNRN_ENABLE_RX3D=OFF \
+  -DCMAKE_INSTALL_PREFIX=$HOME/install
+  -DCMAKE_C_COMPILER=icc \
+  -DCMAKE_CXX_COMPILER=icpc \
+  -DNRN_ENABLE_TESTS=ON
+make -j
```
Comment (olupton) [inline README.md:261]:
```suggestion
cmake --build . --parallel
```

File: README.md [hunk 5]
```diff
@@ -234,6 +234,34 @@ For other errors, please [open an issue](https://github.com/BlueBrain/CoreNeuron
...
+make test
```
Comment (olupton) [inline README.md:262]:
```suggestion
ctest # use --parallel for speed, -R to run specific tests
```

File: README.md [hunk 6]
```diff
@@ -234,6 +234,34 @@ For other errors, please [open an issue](https://github.com/BlueBrain/CoreNeuron
...
+* Integration tests of CoreNEURON
+* Integration tests between NEURON and CoreNEURON
+* [ringtest](https://github.com/neuronsimulator/ringtest) tests with NEURON and CoreNEURON
+* [testcorenrn](https://github.com/neuronsimulator/testcorenrn) tests with NEURON and CoreNEURON
+Some of those tests are going to be also run with various backends in case that those are enabled (for example with GPUs).
+To build NEURON with CoreNEURON and run the tests you need to:
+
+```bash
+cd nrn/build
+cmake .. \
```
Comment (olupton) [inline README.md:253]: Personally I would add `-G Ninja` 😇

File: README.md [hunk 7]
```diff
@@ -234,6 +234,34 @@ For other errors, please [open an issue](https://github.com/BlueBrain/CoreNeuron
...
+  -DNRN_ENABLE_CORENEURON=ON \
+  -DNRN_ENABLE_INTERVIEWS=OFF \
+  -DNRN_ENABLE_RX3D=OFF \
+  -DCMAKE_INSTALL_PREFIX=$HOME/install
+  -DCMAKE_C_COMPILER=icc \
```
Comment (olupton) [inline README.md:258]: I think it normally isn't necessary to specify these explicitly?

**Review by olupton — APPROVED** [review]
LGTM, only minor suggestions you should feel free to ignore :+1:

File: README.md (hunk 6 above)
Comment (iomaganaris) [reply README.md:253]: Haha I think that since this is not mandatory we can keep it aside :stuck_out_tongue_closed_eyes:

File: README.md (hunk 7 above)
Comment (iomaganaris) [reply README.md:258]: Indeed. They were just added above, so I decided to keep them, to avoid any confusion

File: README.md (hunk 4 above)
Comment (iomaganaris) [reply README.md:261]: Should this be changed in other places as well?

File: README.md (hunk 6 above)
Comment (pramodk) [reply README.md:253]: yeah, I would keep this aside (ninja is not always installed on end-users machine, right?)

File: README.md (hunk 4 above)
Comment (pramodk) [reply README.md:261]: no strong opinion - until now everywhere we have make. (if this is preferred way and portable with cmake versions we support, 3.8, then ok to change?)

Comment (iomaganaris) [reply README.md:261]: This is supported in `cmake 3.8` as well. If you think it's fine maybe we should change it everywhere in CoreNEURON and in NEURON as well to be consistent

**Review by pramodk — APPROVED** [review]

**Pull request merged**